ding/framework/middleware/pace_controller.py

```diff
@@ -0,0 +1,62 @@
+from typing import TYPE_CHECKING
+import time
+import math
+if TYPE_CHECKING:
+    from ding.framework import Task, Context
+
+
+def Pace_Controller(
+    task: "Task",
+    theme: str = "unknown",
+    active_state: bool = True,
+    filter_the_similar: bool = True,
+    start_up_delay: float = 0.5,
+    timeout: float = math.inf,
+    emit_option: bool = True
+):
+
+    if start_up_delay > 0:
+        time.sleep(start_up_delay)
+
+    random_event = "pace_control_{}".format(theme)
+    event_received = False
+    all_middleware_names = list(map(lambda fn: getattr(fn, "__name__", type(fn).__name__), task.middleware))
+
+    def _event_received(node_id, middleware_names):
+        nonlocal event_received
+        not_same = map(lambda name: name not in all_middleware_names, middleware_names)
```

Comment:
Concat all names into string, so we don't need to loop over the list every time.

Reply:
Thanks, it is a good idea. But I found a logic bug in it when the middlewares are registered in different sequence. A simpler implementation is realized in the new version.